lib/countdown_animation.dart

```diff
@@ -20,37 +20,36 @@ class CountDownAnimation extends StatefulWidget {
   final Widget child;
   final double strokeWidth;
   final EdgeInsetsGeometry padding;
-  final Function(int) onChanged;
+  final Function(int)? onChanged;
   final int totalNumber;
   final Operation operation;
   final int initialCounterIndex;
   const CountDownAnimation({
-    Key key,
-    @required this.controller,
-    @required this.size,
-    @required this.child,
-    @required this.totalNumber,
-    @required this.initialCounterIndex,
-    @required this.operation,
+    Key? key,
+    required this.controller,
+    required this.size,
+    required this.child,
+    required this.totalNumber,
+    required this.initialCounterIndex,
+    required this.operation,
     this.strokeWidth = 5.0,
     this.padding = EdgeInsets.zero,
     this.onChanged,
     this.alignment = Alignment.center,
     this.progressColor = Colors.red,
     this.backgroundColor = Colors.grey,
-  })  : assert(size != null),
-        super(key: key);
+  }) : super(key: key);
 
   @override
   _CountDownAnimationState createState() => _CountDownAnimationState();
 }
 
 class _CountDownAnimationState extends State<CountDownAnimation>
     with TickerProviderStateMixin {
-  AnimationController countDownAnimationController;
-  double _percentage;
-  double _nextPercentage;
-  int _currentNumber;
+  late AnimationController countDownAnimationController;
+  double? _percentage;
+  double? _nextPercentage;
+  int? _currentNumber;
```

Comment:
You can use `late keyword` for these fields. Now, they are initialized in `initState`. If the timing which they could be `null` is only before `initState` like followings.

```dart
late var _currentNumber = widget.initialCounterIndex;
```